File: jifrozen/최단경로/경로찾기.java
package 최단경로;
import java.util.*;

public class 경로찾기 {
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in);
		int INF=(int)1e9;

		int n=sc.nextInt();
		int[][] graph=new int[n+1][n+1];

		for(int i=0;i<n;i++){
			Arrays.fill(graph[i],INF);
		}

		for(int i=0;i<n;i++){
			for(int j=0;j<n;j++){
				if(sc.nextInt()==1){
					graph[i][j]=1;
				}
			}
		}

		for(int i=0;i<n;i++) {
			for (int j = 0; j < n; j++) {
				for (int l = 0; l < n; l++) {
					graph[j][l] = Math.min(graph[j][l], graph[j][i] + graph[i][l]);
				}
			}
		}

		for(int i=0;i<n;i++){
			for(int j=0;j<n;j++) {
				if(graph[i][j]==INF){
					System.out.print(0+ " ");
				}else{
					System.out.print(1+ " ");
				}
			}
			System.out.println();
		}






	}
}

File: TaeBeomShin/7. 최단경로/경로찾기.py
INF=int(1e9)

n=int(input())

graph=[[INF]*(n+1) for _ in range(n+1)]

for a in range(1,n+1):
    for b in range(1,n+1):
        if a==b:
            graph[a][b]=0
board=[list(map(int,input().split())) for _ in range(n)]

for i in range(n):
    for j in range(n):
        if(board[i][j]==1):
            graph[i+1][j+1]=1

for k in range(1,n+1):
    for a in range(1,n+1):
        for b in range(1,n+1):
            graph[a][b]=min(graph[a][b],graph[a][k]+graph[k][b])

for i in range(1,n+1):
    for j in range(1,n+1):
        if(graph[i][j]!=INF):
            print(1,end=" ")
        else:
            print(0,end=" ")
    print()